applications/nrf_desktop/doc/power_manager.rst

```diff
@@ -3,5 +3,182 @@
+
+For more information about configuration options, check help in the configuration tool.
+
+Implementation details
+**********************
+
+The power manager is started when the :ref:`nrf_desktop_main` module is ready (reported using ``module_state_event``).
+
+When started, it can do the following operations:
+
+* Manage `Power states`_
+* Trigger `Power down`_
+* Handle `Wake-up scenarios`_ from suspended or off states
+
+Power states
+============
+
+The application can be in the following power states:
+
+ * `Idle`_
+ * `Suspended`_
+ * `Off`_
+ * `Error`_
+
+The power manager takes control of the overall operating system power state.
+See the following section for more details about how the application states converts to the system power state.
+
+Idle
+----
+
+In this state, the board is either connected or actively looking for a new connection (advertising).
+The peripherals can be turned on, including LEDs.
+
+The application remains in this state indefinitely if the device is connected to the USB.
+In such case, the operating system will be kept in the ``SYS_POWER_STATE_ACTIVE`` state.
+
+If the device is not connected to a USB, the module counts time elapsed since the last user interaction (that is, since the last HID report sent from the device).
+On timeout, the power manager module sets the application to either the suspended or the off state.
+
+Suspended
+---------
+
+Upon power-down timeout, the power manager will switch the application to the low power state if the following conditions are met:
+
+* The device is connected.
+* The option ``CONFIG_DESKTOP_POWER_MANAGER_STAY_ON`` is selected.
+
+The other modules of the application, if applicable, will turn off the peripherals or switch them to standby to conserve power.
+The operating system will be kept in the ``SYS_POWER_STATE_ACTIVE`` state.
+
+It is assumed that the operating system will conserve power by setting the CPU state to idle whenever possible.
+The established connection is maintained.
+
+Off
+---
+
+Upon power-down timeout, the power manager will switch the application to the deep sleep mode if the following conditions are met:
+
+* The device is disconnected.
+* The option ``CONFIG_DESKTOP_POWER_MANAGER_STAY_ON`` is disabled.
+
+The other modules of the application, if applicable, will turn off the peripherals or switch them to standby to conserve power.
+The operating system will be switched to the ``SYS_POWER_STATE_DEEP_SLEEP_1`` state.
+The devices will be suspended and CPU will be switched to the deep sleep (off) mode.
+
+A device reboot is requited to exit this state.
+
+Error
+-----
+
+In this state, the power manager module checks if any application modules reported an error condition.
+
+When a module switches to the error state, the power manager will put the application to off state after the amount of time defined by ``CONFIG_DESKTOP_POWER_MANAGER_ERROR_TIMEOUT``.
```

Comment:
Ok, I might've got it wrong when editing, or something is not clear here: line 136 says "In this state, the power manager", which suggests it's a state of the power manager.

Then, in line 138, we have "When a module switches to the error state". So I'm confused whether this is the description of the manager's Error state or modules' Error state.

Reply:
I put `a module`, meaning any module in the application. Error state in this context is the internal state of the module passed with `module_state_event`

Reply:
I have made it more explicit now.